### ISSUE TITLE
warning in clustering and error in converting steps

### ISSUE THREAD

**Issue by yagui** [spyking-circus/spyking-circus#237]
I'm having this warning I think during the clustering step

```
python3.6/site-packages/scipy/interpolate/fitpack2.py:227: UserWarning: 
The maximal number of iterations maxit (set to 20 by the program)
allowed for finding a smoothing spline with fp=s has been reached: s
too small.
There is an approximation returned but the corresponding weighted sum
of squared residuals does not satisfy the condition abs(fp-s)/s < tol.
warnings.warn(message)
```

and also an error at the last step
```
-------------------------  Informations  -------------------------
| Last 32 templates are unfitted spikes on all electrodes
------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ariel/virtualenvs/spyking/.spyking/bin/spyking-circus-subtask", line 8, in <module>
    sys.exit(main())
  File "/home/ariel/virtualenvs/spyking/.spyking/lib/python3.6/site-packages/circus/scripts/subtask.py", line 53, in main
    circus.launch(task, filename, nb_cpu, nb_gpu, use_gpu, extension=extension)
  File "/home/ariel/virtualenvs/spyking/.spyking/lib/python3.6/site-packages/circus/__init__.py", line 21, in launch
    module.main(params, nb_cpu, nb_gpu, use_gpu, extension)
  File "/home/ariel/virtualenvs/spyking/.spyking/lib/python3.6/site-packages/circus/converting.py", line 371, in main
    N_tm = write_templates(output_path, params, extension)
  File "/home/ariel/virtualenvs/spyking/.spyking/lib/python3.6/site-packages/circus/converting.py", line 208, in write_templates
    waveform = io.get_stas(params, times_i=spikes, labels_i=np.ones(len(spikes)), src=elec, neighs=[elec], nodes=nodes, mean_mode=True)
  File "/home/ariel/virtualenvs/spyking/.spyking/lib/python3.6/site-packages/circus/shared/files.py", line 134, in get_stas
    local_chunk = data_file.get_snippet(time - offset, duration, nodes=nodes)
  File "/home/ariel/virtualenvs/spyking/.spyking/lib/python3.6/site-packages/circus/files/datafile.py", line 462, in get_snippet
    return self.get_data(0, chunk_size=length, padding=(local_time, local_time), nodes=nodes)[0]
  File "/home/ariel/virtualenvs/spyking/.spyking/lib/python3.6/site-packages/circus/files/datafile.py", line 471, in get_data
    return self.read_chunk(idx, chunk_size, padding, nodes), self.t_start + idx*chunk_size
  File "/home/ariel/virtualenvs/spyking/.spyking/lib/python3.6/site-packages/circus/files/raw_binary.py", line 53, in read_chunk
    local_chunk = self.data[t_start*self.nb_channels:t_stop*self.nb_channels]
  File "/home/ariel/virtualenvs/spyking/.spyking/lib/python3.6/site-packages/numpy/core/memmap.py", line 331, in __getitem__
    res = super(memmap, self).__getitem__(index)
TypeError: slice indices must be integers or None or have an __index__ method
```
Here is the complete log
[log.txt](https://github.com/spyking-circus/spyking-circus/files/4404724/log.txt)
And here is the params file
[continuous.params.txt](https://github.com/spyking-circus/spyking-circus/files/4404730/continuous.params.txt)

Thanks

**Comment by yger** [comment]
Are you sure the data are correctly loaded? Because this error message means that the smoothing failing, and if it does so, it may be because the data are not properly loaded. Did you try to look at the preview first?

**Comment by yagui** [comment]
You are right. The data file was corrupted. (I don't know why :( )
I tried with a new data file and I did not get the warning but I did have the error in the converting step.
I found that it is related to the option
export_all = True
if I return it to False, it does not fail

**Comment by yger** [comment]
Ok, at least probem partially solved. However, what is your error when export_all is set to True?

**Comment by yger** [comment]
This has been fixed in the master branch (0.9.7). Thanks for reporting that!
